src/odemis/gui/model/main_gui_data.py

```diff
@@ -250,10 +250,11 @@ def __init__(self, microscope):
             for c in components:
                 if c.role is None:
                     continue
+
+                comps_with_role.append(c)
```

Comment:
Why is this now outside the try block?

Reply:
That's the core of the change: we now observe *every* component, not just the ones which are used by the GUI explicitly.